pkgs/apps/gfn0/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv
+, lib
+, cmake
+, gfortran
+, fetchFromGitHub
+, blas
+, lapack
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gfn0";
+  version = "unstable-22-07-2023";
+
+  src = fetchFromGitHub {
+    owner = "pprcht";
+    repo = pname;
+    rev = "b0d68ec6b44a176db6c3684d7ccc9776e9a50394";
+    hash = "sha256-257XGz5ZosPnbWjTgM2Bt7hH09ZQkTaW5Vu7udMSIhI=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gfortran
+  ];
+
+  buildInputs = [
+    blas
+    lapack
+  ];
+
+  postInstall = ''
+    substituteInPlace $out/lib/cmake/${pname}/${pname}-*.cmake \
+      --replace "libgfn0.a" "libgfn0.${stdenv.hostPlatform.extensions.sharedLibrary}"
```

Comment:
This replaces a static library name with a shared one. Does this yield the correct result?

Reply:
Yes, the Cmake file hardcodes the `.a` suffix which is not true if we build dynamically as given by the `cmakeFlags`.

Reply:
Ah right, it should be conditional ... :grimacing: